Add state modules for Zabbix templates, actions and valuemaps

slivik opened the pull request:
### What does this PR do?
Adds functions to existing Zabbix module, test and docs related to newly added state modules for Zabbix templates, actions and value maps.

### What issues does this PR fix or reference?
None

### Tests written?

Yes

### Commits signed with GPG?

Yes

slivik committed:
Add state modules for Zabbix templates, actions and valuemaps

slivik committed:
Merge branch 'develop' into zabbix_template_state

cachedout commented:
@jirikotlin Can you look some of this over? I know you've done a lot of work in this area.

mchugh19 commented on salt/modules/zabbix.py:
```diff
@@ -2100,10 +2333,86 @@ def run_query(method, params, **connection_args):
+    '''
+    .. versionadded:: 2017.7
+
+    Imports Zabbix configuration sepcified in file to Zabbix server.
```
Spelling: specified

mchugh19 commented on salt/modules/zabbix.py:
```diff
@@ -846,8 +1070,8 @@ def host_create(host, groups, interfaces, **connection_args):
         salt '*' zabbix.host_create technicalname 4
-        interfaces='{type: 1, main: 1, useip: 1, ip: "192.168.3.1", dns: "", port: 10050}'
-        visible_name='Host Visible Name' inventory_mode=0 inventory='{"alias": "something"}'
+        interfaces='{if_type: 1, main: 1, useip: 1, ip_: "192.168.3.1", dns: "", port: 10050}'
```
What is the gain from the change in user facing syntax? This line also clobbers an example of setting inventory information.

slivik replied:
Sorry, my mistake. Will fix it in next commit.

mchugh19 replied:
The Zabbix API uses the following for interfaces in host.create:
```
"interfaces": [
            {
                "type": 1,
                "main": 1,
                "useip": 1,
                "ip": "192.168.3.1",
                "dns": "",
                "port": "10050"
            }
        ]
```

Why the change to if_type and ip_? This will affect user configurations and existing states, so is the rename necessary?

slivik replied:
You are right, it was a typo. Please, see my last [commit](https://github.com/saltstack/salt/pull/47398/commits/612f2a5f207db90254ff6259e01356d017ab62e3#diff-48649b036902f9e3cd75a774e29e1b91R1072) where it is fixed.

slivik committed:
fixed pylint and typos

slivik committed:
fixed pylint and typos
Merge branch 'zabbix_template_state' of https://github.com/slivik/salt into zabbix_template_state

mchugh19 commented on salt/modules/zabbix.py:
```diff
@@ -1466,7 +1689,7 @@ def hostinterface_get(hostids, **connection_args):
 
 
-def hostinterface_create(hostid, ip, dns='', main=1, type=1, useip=1, port=None, **connection_args):
+def hostinterface_create(hostid, ip_, dns='', main=1, if_type=1, useip=1, port=None, **connection_args):
```
Should this still be ip_ and if_type?

slivik replied:
Yes, in this case it is correct since Salt coding style (and pylint) says at least three characters for variable name and "type" is reserved Python keyword.

slivik committed:
Add function to zabbix_template state module.
